modules/openxr/scene/openxr_composition_layer.cpp

```diff
@@ -285,6 +293,40 @@ void OpenXRCompositionLayer::_notification(int p_what) {
 	}
 }
 
+void OpenXRCompositionLayer::_get_property_list(List<PropertyInfo> *p_property_list) const {
+	List<PropertyInfo> extension_properties;
+	for (OpenXRExtensionWrapper *extension : OpenXRAPI::get_registered_extension_wrappers()) {
```

Comment:
I wonder if we should constrain the list; i.e: instead of checking all extension wrappers, we only check registered viewport composition layer extension wrapper.

I don't have a strong opinion either way though.

Reply:
Well, we _could_ specially register some extensions that participate in this API, like we do with `register_composition_layer_provider()`. However, the difference is that those composition layer providers do their thing every single frame, so it's important to loop over the fewest extensions.

Whereas the APIs added in this PR are run very infrequently, for example, the result of `_get_property_list()` will be cached, and the `set_viewport_composition_layer_and_get_next_pointer()` will only be run if one of the extension properties changes. So, I think in this case it's fine to loop over them all, and makes things simpler by not requiring another registration.